lib/saml_idp/xml_security.rb

```diff
@@ -209,7 +213,11 @@ def validate_doc_embedded_signature(base64_cert, soft = true)
             inclusive_namespaces
           )
 
-          digest_algorithm = digest_method_algorithm(ref, sig_namespace_hash)
+          digest_algorithm = if no_namespace_enabled
+              digest_method_algorithm(ref, sig_namespace_hash)
+            else
+              algorithm(REXML::XPath.first(ref, '//ds:DigestMethod'))
+            end
```

Comment:
spacing is goofy here, I'd expect aligned with `digest_algorithm` or with `if`?

this is my personal fave
```suggestion
          digest_algorithm = if no_namespace_enabled
            digest_method_algorithm(ref, sig_namespace_hash)
          else
            algorithm(REXML::XPath.first(ref, '//ds:DigestMethod'))
          end
```

Reply:
updated in https://github.com/18F/saml_idp/pull/104/commits/be4c97e4eccd4596853053b9100f50420d552d4b